### DISCUSSION TITLE
Is the document to be validated loaded completely in memory

### DISCUSSION THREAD

**Discussion by rmuller** [networknt/json-schema-validator#995] [Q&A] [answered]
It is not clear to me if the document to be validated loaded completely in memory or it is processed in a streaming mode.
If it is completely loaded in memory, is there a way to enable a streaming mode? If not, any suggestions how to implement this?

**Comment by justin-tay** [comment, marked as answer]
Yes it's completely loaded into memory with a call to Jackson's `ObjectMapper.readTree`. The api is such that it calls validate with the `JsonNode` of the current instance data and the `JsonNode` of the root instance data.

It's not possible to enable a streaming mode. I don't really have an idea of how you can implement this generically.

If what you have is a document with an array of objects, you could stream and process each object individually with a json schema for each object.

**Comment by rmuller** [comment]
Indeed the JSON file exists of a header object and an array with a huge amount of array elements.
The solution you propose is the same i was thinking of (we do the same in the XML world). Wanted to be sure this is also in this case the best approach. Thanks for your quick reply!
